testdata/go/v3/monitoring/memcached-operator/monitoring/metricsdocs/metricsdocs.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/example/memcached-operator/monitoring"
+)
+
+// please run "make generate-metricsdocs" to run this tool and update metrics documentation
+const (
+	title      = "# Operator Metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by this operator.\n" +
+		"The metrics documentation is auto-generated by the utility tool \"monitoring/metricsdocs\" and reflects all of the metrics that are exposed by the operator.\n\n"
+
+	KVSpecificMetrics = "## Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run \"make generate-metricsdocs\" to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change \"monitoring/metricsdocs\" according to your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+// TODO: scaffolding these helpers with operator-lib: https://github.com/operator-framework/operator-lib.
```

Comment:
@machadovilaca Could we remove this file out of the PR for now? I am concerned that this is generic and would it would not value in scaffolding it for every project. Can we remove this, and have documentation scaffolded only on the type of metric in memcached project for now.

Reply:
@varshaprasad96 This is the core of the PR. This is the metrics doc generator...
It creates the metrics documentation. An example output for it would be https://github.com/machadovilaca/kubevirt/blob/main/docs/metrics.md.

See how long this file is due to the amount of metrics that this operator has.
Having an example for auto generating this based on the metrics code is very helpful for both operators developers and its users.

Reply:
We should allow the users to make changes based on their needs, so I don't this this would be a good fit as a library. Having it as documentation would miss the point.

Reply:
@sradco I understand that the doc needs to be inside the project, and this is the generator for the same. My worry is that the common code that is definition of `Metrics` and helpers like `writeToFile` are of no added value to the operator author directly. In the sense these need not be scaffolded for every project. They can be called directly from a library helper.

Reply:
We decided to create a lib in a follow-up pr. 
closing this conversation.